packages/stark-ui/src/modules/pretty-print/components/pretty-print.component.ts

```diff
@@ -0,0 +1,169 @@
+import { Component, HostBinding, Inject, Input, OnChanges, OnInit, SimpleChanges, ViewEncapsulation } from "@angular/core";
+
+/* tslint:disable:no-duplicate-imports no-import-side-effect */
+import * as Prism from "prismjs";
+import { LanguageDefinition } from "prismjs";
+import "prismjs/components/prism-typescript.min.js";
+import "prismjs/components/prism-sql.min.js";
+import "prismjs/components/prism-json.min.js";
+import "prismjs/components/prism-scss.min.js";
+/* tslint:enable */
+
+import { STARK_LOGGING_SERVICE, STARK_ROUTING_SERVICE, StarkLoggingService, StarkRoutingService } from "@nationalbankbelgium/stark-core";
+
+/**
+ * Name of the component
+ */
+const componentName: string = "stark-pretty-print";
+
+/**
+ * The prefix used in the PrismJs css classes
+ */
+const prismClassPrefix: string = "language-";
+
+/**
+ * A reference to the prettyData library
+ */
+const prettyData: any = require("pretty-data").pd;
+
+/**
+ * The code languages that are supported by the Stark-Pretty-Print component
+ */
+export type StarkPrettyPrintFormat = "css" | "scss" | "html" | "xml" | "json" | "sql" | "javascript" | "typescript";
+
+/**
+ * Component to format and highlight code like HTML, CSS, Typescript...
+ * Can be used to display code examples
+ */
+@Component({
+	selector: "stark-pretty-print",
+	templateUrl: "./pretty-print.component.html",
+	encapsulation: ViewEncapsulation.None
+})
+export class StarkPrettyPrintComponent implements OnChanges, OnInit {
+	public prettyString: string;
```

Comment:
As a convention we should put the `@Input` properties first and then the internal properties...